test/crawlers/statistics.test.js

```diff
@@ -9,18 +11,135 @@ describe('Statistics', () => {
+            expect(n1.id).toEqual(1);
+            expect(n2.id).toEqual(2);
+            expect(Statistics.id).toEqual(3);
+        });
+
+        test('should persist the state to KV and load again', async () => {
+            const startedAt = 1000;
+            clock.tick(startedAt);
+            stats.startJob(0);
+            clock.tick(100);
+            stats.finishJob(0);
+
+            await stats.startLogging();
+            await stats.persistState();
+
+            const state = await stats.keyValueStore.getValue(stats.persistStateKey);
+
+            expect(state).toEqual({
+                jobRetryHistogram: [1],
+                finishedJobs: 1,
+                failedJobs: 0,
+                totalJobDurationMillis: 100,
+                startedAt,
+            });
+
+            await stats.stopLogging();
+            stats.reset();
+
+            expect(stats.toJSON()).toEqual({
+                jobRetryHistogram: [],
+                finishedJobs: 0,
+                failedJobs: 0,
+                totalJobDurationMillis: 0,
+                startedAt: 0,
+            });
+
+            await stats.startLogging();
+
+            stats.startJob(1);
+            clock.tick(100);
+            stats.finishJob(1);
+
+            expect(stats.toJSON()).toEqual({
+                jobRetryHistogram: [2],
+                finishedJobs: 2,
+                failedJobs: 0,
+                totalJobDurationMillis: 200,
+                startedAt,
+            });
+
+            clock.tick(1000);
+
+            expect(stats.getCurrent()).toEqual({
+                avgDurationMillis: 100,
+                perMinute: 100,
+                finished: 2,
+                failed: 0,
+                retryHistogram: [2],
+            });
+        });
+
+        test('should remove persist state event listener', async () => {
+            await stats.startLogging();
+            expect(events.listenerCount(ACTOR_EVENT_NAMES_EX.PERSIST_STATE)).toEqual(1);
+            await stats.stopLogging();
+
+            expect(events.listenerCount(ACTOR_EVENT_NAMES_EX.PERSIST_STATE)).toEqual(0);
+            await stats.startLogging();
+            expect(events.listenerCount(ACTOR_EVENT_NAMES_EX.PERSIST_STATE)).toEqual(1);
+            stats.reset();
+
+            expect(events.listenerCount(ACTOR_EVENT_NAMES_EX.PERSIST_STATE)).toEqual(0);
+        });
+
+        test('on persistState event', async () => {
+            stats.startJob(0);
+            clock.tick(100);
+            stats.finishJob(0);
+
+            await stats.startLogging(); // keyValueStore is initialized here
+
+            events.emit(ACTOR_EVENT_NAMES_EX.PERSIST_STATE);
+
+            clock.restore(); // restore clock so setInterval works again
+
+            const state = await new Promise((resolve) => {
+                const interval = setInterval(async () => {
+                    const persisted = await stats.keyValueStore.getValue(stats.persistStateKey);
```

Comment:
This is interesting. I would probably mock the `setValue` call before the clock ticks, but this works as well.